src/sourcerer/infrastructure/storage_provider/services/s3.py

```
@@ -218,26 +218,28 @@ def delete_storage_item(self, storage: str, key: str) -> None:
             UploadStorageItemsException: If an error occurs while uploading the item
         """
         try:
-            self.client.upload_file(source_path, storage, dest_path or source_path.name)
+            dest_path = str(Path(storage_path or '') / (dest_path or source_path.name))
+            self.client.upload_file(source_path, storage, dest_path)
         except Exception as ex:
             raise UploadStorageItemsException(str(ex)) from ex
 
     def download_storage_item(
-        self, storage: str, key: str, progress_callback: Callable | None = None
+            self, storage: str, key: str, progress_callback: Callable | None = None
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)